content/en/docs/architecture/ci-operator-internals/testing.md

```diff
@@ -0,0 +1,425 @@
+
+{{< highlight bash >}}
+$ ci-operator --namespace bbguimaraes --config test.yaml --target test
+INFO[2022-03-10T14:41:33Z] unset version 0
+INFO[2022-03-10T14:41:33Z] Resolved source https://github.com/openshift/ci-tools to master@ef415037
+INFO[2022-03-10T14:41:35Z] Using namespace https://console.build01.ci.openshift.org/k8s/cluster/projects/bbguimaraes
+INFO[2022-03-10T14:41:35Z] Running [input:fedora], test
+INFO[2022-03-10T14:41:37Z] Tagging ci/fedora:latest into pipeline:fedora.
+INFO[2022-03-10T14:41:37Z] Executing test test
+INFO[2022-03-10T14:41:49Z] Ran for 16s
+{{< / highlight >}}
+
+## Target
+
+The last command execution included a few new concepts.  Starting with the
+command line, there is now a new argument: `--target`.  It limits the execution
+to just our test; we will come back to it once the rest of the output is
+analyzed.  Right after the familiar line about the `JOB_SPEC`, we see our test
+namespace was successfully selected.  This line is useful to verify that the
+correct namespace is used, but even more so in CI jobs, where the name of the
+test namespace is automatically generated and not known until the execution of
+the job begins.
+
+The next line is the sequence of execution steps that will be performed.  It is
+a comma-separated list of names, each of which corresponds to one task performed
+by `ci-operator` for a given test (these are called "steps" internally and are
+described in [this page][steps]).  Here, it shows the input image declared in
+`base_images` and the test, which are discrete steps in the execution (an
+[`InputImageTagStep`]({{< relref "steps#inputimagetagstep" >}}) and a
+[`TestStep`]({{< relref "steps#teststep" >}}), respectively).
+
+In reality, steps are organized into an execution _graph_, as described in the
+[introduction][index], where some depend on others for their own execution.  In
+the current example, the test cannot be scheduled before the image is imported:
+the test is _dependent_ on the image import.  One useful property of the list in
+the output is that it is a _topological ordering_ of the graph, i.e. dependents
+always come before the steps they depend on.  As a one-dimensional projection of
+a partial order, it is of limited usefulness in more complicated graphs with
+multiple dependencies between nodes, but it gives a general idea of the order of
+execution.
+
+The next two lines are each output by their respective step when it is executed.
+The image import describes its input and output (the meaning of `pipeline` will
+be explained in a [later section]({{< relref "#pipeline" >}})) and the test
+simply notifies that it is being executed.  Finally, the total execution time is
+displayed right at the end of the execution of `ci-operator`.
+
+Why was the `--target` argument necessary?  Remember that `ci-operator`'s
+defaults are optimized for its use in CI jobs.  As such, it implicitly includes
+common job tasks in the execution graph:
```

Comment:
"common job tasks": What are they?

I thought it would list _all_ steps in the graph without `--target`. No?

Reply:
The extra ones in the `Running` line below: `[output-images]`, `[images]`, `src`.  They are not exactly "all steps", since there are others we choose not to include.  Building the `src` image in this case, for example, is an arbitrary decision since it's "commonly" done.  This is mostly a curiosity, since all jobs have a `--target` argument.

Reply:
:TIL